test/noexcept_task_test.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/coroutine.hpp>
+
+#if !UNIFEX_NO_COROUTINES
+
+#include <atomic>
+
+#include <unifex/static_thread_pool.hpp>
+#include <unifex/when_all.hpp>
+#include <unifex/sync_wait.hpp>
+#include <unifex/task.hpp>
+#include <unifex/single_thread_context.hpp>
+#include <unifex/just_error.hpp>
+#include <unifex/stop_if_requested.hpp>
+
+
+#include <gtest/gtest.h>
+
+using namespace unifex;
+
+UNIFEX_TEMPLATE(typename Scheduler)
+  (requires scheduler<Scheduler>)
+nothrow_task<void> child(Scheduler s, std::atomic<int>& x) {
+  co_await schedule(s);
+  ++x;
+}
+
+UNIFEX_TEMPLATE(typename Scheduler)
+  (requires scheduler<Scheduler>)
+nothrow_task<void> example(Scheduler s, std::atomic<int>& x) {
+  ++x;
+  co_await when_all(child(s, x), child(s, x));
+}
+
+nothrow_task<void> nothrowThrowsException() {
+  throw std::exception{};
+  co_return;
+}
+
+nothrow_task<void> nothrowJustError() {
+  co_await unifex::just_error(std::exception_ptr());
+}
+
+task<void> nothrowTaskBody() {
+  co_await nothrowJustError();
+}
+
+task<int> foo() {
+  co_await stop(); // sends a done signal, unwinds the coroutine stack
+  ADD_FAILURE();
+  co_return 42;
+}
+
+nothrow_task<int> bar() {
+  try {
+    co_await foo();
+    ADD_FAILURE();
+  }
+  catch (...) {
+    ADD_FAILURE();
+  }
+  co_return -1;
```

Comment:
This made me realize we should have a test case that proves that you can safely catch exceptions _within_ a `nothrow_task`; i.e. if `co_await foo()` throws an exception, you ought to be able to catch it here and handle it without the `nothrow_task` machinery somehow interrupting things and terminating the process.